### ISSUE TITLE
Extrapolation with multiple images - UsingContextParallelEncoder3D class - SAT

### ISSUE THREAD

**Issue by antragoudaras** [THUDM/CogVideo#571]
Great work overall, 
@zRzRzRzRzRzRzR 

I am trying to utilize the `CogVideoX-1.5_5B_I2V` model to "reconstruct" part of the motion of a simple object in a video by conditioning on more than one consecutive frame (extrapolation):  
---

Originally, in `sat/sample_video.py`, a single frame of size `(1, 3, 1, H, W)` is used to condition the model. This frame is passed through the `ContextParallelEncoder3D` (after applying some transformations to the height and width of the image), resulting in an encoded representation of size `(1, 1, C, H/8, W/8)`. 

The remaining 21 frames (`args.sampling_num_frames`) are concatenated along the time axis using `torch.zeros`, forming a `torch.Tensor` of size `(1, 22, C, H, W)`. After 50 denoising steps with the `VPSDEDPMPP2MSampler` the `samples_z` output maintains the same dimensions `(1, 22, C, H, W)`. Finally, the `ContextParallelDecoder3D` decompresses the spatial dimensions by a factor of 8 and the temporal dimension by a factor of 4 (from 22 → 48), aligning with the description in the paper.

I noticed, however, that the decoder skips (does not decode) the latent `sample_z` corresponding to the image used to condition the model, as indicated in [this line](https://github.com/THUDM/CogVideo/blob/main/sat/diffusion_video.py#L183). I reviewed the related issue [#524](https://github.com/THUDM/CogVideo/issues/524), but I believe that this is not the case. Debugging the code suggests that the latent at the zero position indeed corresponds to the conditioning image.

In my first attempt to condition the model on multiple consecutive frames, I aimed to capture part of the object's motion, expecting CogVideo to resume the generation while keeping the text prompt simple. Alternatively, I considered leveraging your `CogVLM2` model to caption my videos. Initially, I encoded each image separately using `encode_first_stage` from `SATVideoDiffusionEngine`, as done in [single image scenarios](https://github.com/THUDM/CogVideo/blob/main/sat/sample_video.py#L169). I then stacked each image's representation along the time axis consecutively and filled the remaining `T - num_of_images` with zeros.  
---

Video generated with `CogVideoX-1.5-I2V` using 11 frames:  

https://github.com/user-attachments/assets/525cf347-e11d-4d47-b673-9b110a80dbef

---
However, I realized that this approach does not leverage compression along the temporal dimension. Instead, I tried encoding all the frames simultaneously. Specifically, I fed the `ContextParallelEncoder3D` with a tensor of `N` images of size `(1, 3, N, H, W)`. This method produced an encoded representation of size `(1, N/4, C, H/8, W/8)`. Using this approach, I was able to condition the model on a larger number of consecutive frames (e.g., 44 frames compressed to 11). The remaining frames were filled with zeros as before.  
---
Video generated following this approach using 44 frames:  

https://github.com/user-attachments/assets/28d8c16d-e749-4aa5-b2d6-f895e68fae05

I got a bit confused while reviewing the `train_cogvideox_image_to_video_lora.py` script, which utilizes the Diffusers library. It seems to suggest that some form of noise needs to be encoded separately from the video frames used for fine-tuning. 

However, this approach doesn’t seem to align with the inference code in `sat/sample_video.py`.
Do you have any suggestions or guidance on how to achieve a more faithful reconstruction of the original videos? Specifically, I’m looking for advice on conditioning the model on multiple consecutive frames or optimizing the encoding and decoding processes.
---

**Comment by zRzRzRzRzRzRzR** [comment]
The official diffusers fine-tuning has not implemented the t patch, as you have observed. This is because we are currently using fine-tuning in cogvieox-factory, so the script here is for cogvideox1.0, not 1.5. Due to limited manpower, we are currently unable to invest labor in the research of fine-tuning. I believe that some parts of cogvieox-factory can work normally with the t patch now?

**Comment by antragoudaras** [comment]
Thanks for the prompt response, I am already aware of the cogvideox-factory and I was planning to use at same point. My **question is not on fine-tuning the model**, but rather about **using multiple frames as guidance during inference** using the `sat/sample_video.py` script, as to achive the best "reconstruction results".

To clarify, I am not currently using the diffusers implementation. Instead, I am working with the SwissArmyTransformer (SAT) framework developed as by your team, which, as I understand, was primarily used for the CogVideoX paper before integrating with the diffusers for efficiency improvements.
